test/unit/pym_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class PymAwareBuilder < Object
+  def render(delta)
+    RichText::HTML.new(context: self).render(delta).inner_html
+  end
+
+  def build_pym(el, op)
+    pym = op.value[:pym]
+    el.name = 'div'
+    el.add_class('pym')
+    el.add_child(%(<p data-pym-src="#{ pym[:url] }"><a href="#{ pym[:url] }">Don't see the graphic? Click here.</a></p>))
+    el.add_child(%(<script></script>))
+    script = el.children.last
```

Comment:
nice! it looks like [#add_child](https://github.com/sparklemotion/nokogiri/blob/c0bce3cfd4546f1168648ccb9e2dd16c4fe6f105/lib/nokogiri/xml/node.rb#L152) returns a Node or NodeSet (in this case it's a node set for whatever reason).  I could see using that return value a more stable way of accessing this new child node.
```suggestion
    script = el.add_child(%(<script></script>)).first
```

Reply:
i tried that, but it didn't work like I expected. I think it returned `el` not the child i added. but i'll leave that for the sbn code to figure out.